src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -237,37 +244,61 @@ private void handleDescriptionElement(ImmutableList.Builder<KeyValueLabelMetadat
 
     private void handleVariableElement(ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder,
             XMLStreamReader xmlStreamReader) {
+
         handleElementWithMultipleValues(keyValueMapBuilder,
                                         ATTRIBUTE_VARIABLE_LABEL,
                                         ATTRIBUTE_VARIABLE_NAME,
                                         ATTRIBUTE_VARIABLE_VALUE,
-                                        xmlStreamReader);
+                                        xmlStreamReader,
+                                        true);
+        //for variables model
+        handleElementWithMultipleValues(keyValueMapBuilder,
+                                        ATTRIBUTE_VARIABLE_MODEL_LABEL,
+                                        ATTRIBUTE_VARIABLE_NAME,
+                                        ATTRIBUTE_VARIABLE_MODEL,
+                                        xmlStreamReader,
+                                        false);
+
     }
 
     private void handleElementWithMultipleValues(ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder,
             String attributeLabel, String attributeNameForKey, String attributeNameForValue,
-            XMLStreamReader xmlStreamReader) {
-        String[] key = new String[1];
-        String[] value = new String[1];
-
-        iterateOverAttributes((attributeName, attributeValue) -> {
-            if (attributeName.equals(attributeNameForKey)) {
-                key[0] = attributeValue;
-            } else if (attributeName.equals(attributeNameForValue)) {
-                value[0] = attributeValue;
+            XMLStreamReader xmlStreamReader, boolean allowEmptyValues) {
+        List<String> strings = orderedSearch(xmlStreamReader, attributeNameForKey, attributeNameForValue);
+
+        String key = strings.get(0);
+        String value = strings.get(1);
+
+        if (checkIfNotNull(key, value)) {
+            if (allowEmptyValues || checkIfNotEmpty(key, value)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)